pydicom/dataelem.py

```diff
@@ -433,13 +433,22 @@ def value(self) -> Any:
     @value.setter
     def value(self, val: Any) -> None:
         """Convert (if necessary) and set the value of the element."""
+        # Ignore backslash characters in these VRs, based on:
+        # * Which str VRs can have backslashes in Part 5, Section 6.2
+        # * All byte VRs
+        exclusions = [
+            'LT', 'OB', 'OD', 'OF', 'OL', 'OV', 'OW', 'ST', 'UN', 'UT',
+            'OB/OW', 'OW/OB', 'OB or OW', 'OW or OB',
+        ]
+
         # Check if is a string with multiple values separated by '\'
         # If so, turn them into a list of separate strings
         #  Last condition covers 'US or SS' etc
-        if isinstance(val, (str, bytes)) and self.VR not in \
-                ['UT', 'ST', 'LT', 'FL', 'FD', 'AT', 'OB', 'OW', 'OF', 'SL',
-                 'SQ', 'SS', 'UL', 'OB/OW', 'OW/OB', 'OB or OW',
-                 'OW or OB', 'UN'] and 'US' not in self.VR:
+        if (
+            isinstance(val, (str, bytes))
+            and self.VR not in exclusions
+            and 'US' not in self.VR
```

Comment:
Either we allow to assign `bytes` to numeric VRs, in which case all these VRs shall be added to exclusions, or not, in which case this should not be needed.

Reply:
Looking at the rest of the converter code and using US as the example, if it was passed in using bytes it would end up unconverted as raw bytes.
```python
>>> from pydicom import Dataset
>>> ds = Dataset()
>>> ds.Rows = b'\x00\x01'
>>> ds.Rows
b'\x00\x01'
>>> ds["Rows"].VR
'US'
```
But having said that I think I'll just put back the other VRs to be safe

Reply:
I agree. You could just add `US` to the exclusions, though - I don't see any reason why it should be handled separately.

Reply:
I think it was a shortcut so the various 'US or ...' ambiguous VRs were also included.

Reply:
You are right of course - completely missed that. I shouldn't do any reviewing in this heat...